lib/src/Uncertainty/Distribution/VonMisesFactory.cxx

```diff
@@ -0,0 +1,167 @@
+  : DistributionFactoryImplementation()
+  , circular_(circular)
+{
+  // Nothing to do
+}
+
+/* Virtual constructor */
+VonMisesFactory * VonMisesFactory::clone() const
+{
+  return new VonMisesFactory(*this);
+}
+
+struct VonMisesFactoryParameterConstraint
+{
+  /** Constructor from a sample and a derivative factor estimate */
+  VonMisesFactoryParameterConstraint(const Scalar logMeanCos)
+    : logMeanCos_(logMeanCos)
+  {
+    // Nothing to do
+  };
+
+  Point computeConstraint(const Point & parameter) const
+  {
+    const Scalar kappa = parameter[0];
+    if (!(kappa > 0.0)) throw InvalidArgumentException(HERE) << "Error: the kappa parameter must be positive.";
+    return Point(1, SpecFunc::DeltaLogBesselI10(kappa) - logMeanCos_);
+  }
+
+  Scalar logMeanCos_;
+};
+
+
+/* Here is the interface that all derived class must implement */
+
+Distribution VonMisesFactory::build(const Sample & sample) const
+{
+  return buildAsVonMises(sample).clone();
+}
+
+Distribution VonMisesFactory::build(const Point & parameters) const
+{
+  return buildAsVonMises(parameters).clone();
+}
+
+Distribution VonMisesFactory::build() const
+{
+  return buildAsVonMises().clone();
+}
+
+VonMises VonMisesFactory::buildAsVonMises(const Sample & sample) const
+{
+  const UnsignedInteger size = sample.getSize();
+  if (size == 0) throw InvalidArgumentException(HERE) << "Error: cannot build a VonMises distribution from an empty sample";
+  if (sample.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: can build a VonMises distribution only from a sample of dimension 1, here dimension=" << sample.getDimension();
+
+  // In the case of circular data
+  Scalar mu = 0.0;
+  Scalar ratio = 0.0;
+  if (circular_)
+    {
+      Scalar sumCosXk = 0.0;
+      Scalar sumSinXk = 0.0;
+      for (UnsignedInteger i = 0; i < size; ++i)
+        {
+          sumCosXk += std::cos(sample(i, 0));
+          sumSinXk += std::sin(sample(i, 0));
+        }
+      sumCosXk /= size;
+      sumSinXk /= size;
+      mu = std::atan2(sumSinXk, sumCosXk);
+      ratio = sqrt((size / (size - 1.0)) * (sumCosXk * sumCosXk + sumSinXk * sumSinXk - 1.0 / size));
+    }
+  // Use two moments: mu=E[X]
+  else
+    {
+      // Mu estimate is the sample mean
+      mu = sample.computeMean()[0];
+      for (UnsignedInteger i = 0; i < size; ++i)
+        ratio += std::cos(sample(i, 0) - mu);
+      ratio /= size;
+    }
+  LOGDEBUG(OSS() << "mu=" << mu);
+  // Kappa estimate is based on E[cos(X-mu)]=I0(kappa)/I1(kappa)
+  // We work on log scale to avoid overflow in I0 and I1
+  if (!(ratio > 0.0)) throw InvalidArgumentException(HERE) << "Error: cannot build a VonMises distribution if ratio is nonpositive, here ratio=" << ratio;
+  LOGDEBUG(OSS() << "ratio=" << ratio);
+  VonMisesFactoryParameterConstraint constraint(std::log(ratio));
+  const Function f(bindMethod<VonMisesFactoryParameterConstraint, Point, Point>(constraint, &VonMisesFactoryParameterConstraint::computeConstraint, 1, 1));
```

Comment:
hmmm it is better to avoid the `bindMethod` and implement easily a `EvaluationImplementation`

Reply:
There is no actual performance issue here, but I will change it. If I remember well it is not so easy to do the job through inheritance from EvaluationImplementation (constructor, operator() but also clone() and a few others...)

Reply:
Agreed with performance. IMHO we should drop this `class`